examples/contour/main.js

```diff
@@ -0,0 +1,97 @@
+// Run after the DOM loads
+$(function () {
+  'use strict';
+
+  // Define a function we will use to generate contours.
+  function makeContour(data, layer) {
+    /* There are two example data sets.  One has a position array which
+     * consists of objects each with x, y, z values.  The other has a values
+     * array which just has our contour values. */
+    var contour = layer.createFeature('contour')
+      .data(data.position || data.values)
+      .style({
+        opacity: 0.75
+      })
+      .contour({
+        gridWidth: data.gridWidth,
+        gridHeight: data.gridHeight,
+        /* The color range doesn't have to be linear:
+        rangeValues: [0, 25, 50, 75, 100, 125, 250, 500, 750, 2000],
+         */
+        /* Or, you could plot iso-contour lines using a varying opacity:
+        rangeValues: [100, 100, 200, 200, 300, 300, 400, 400, 500, 500],
+        opacityRange: [1, 0, 1, 0, 1, 0, 1, 0, 1],
+         */
+        /* You can make smooth contours instead of stepped contours:
+        stepped: false,
+         */
+        min: 0
+      });
+    if (data.position) {
+      contour
+      .position(function (d) { return {x: d.x, y: d.y, z: d.z}; })
+      .style({
+        value: function (d) { return d.z > -9999 ? d.z : null; }
+        /* You can get better contours if you set a minimum value and set
+         * sea locations to a small negative number:
+        value: function (d) { return d.z > -9999 ? d.z : -10; }
+         */
+      });
+    } else {
+      contour
+      .style({
+        value: function (d) { return d > -9999 ? d : null; }
+      })
+      .contour({
+        /* The geometry can be specified using 0-point coordinates and deltas
+         * since it is a regular grid. */
+        x0: data.x0, y0: data.y0, dx: data.dx, dy: data.dy
```

Comment:
Yes.  See the documentation at the end of
https://github.com/OpenGeoscience/geojs/blob/contour-feature/src/core/contourFeature.js

If there is a better or more standard way we have of documenting the
keywords that each feature accepts, I'd be happy to add it there, too.

On Tue, Jun 2, 2015 at 2:22 PM, Aashish Chaudhary <notifications@github.com>
wrote:

> In examples/contour/main.js
> <https://github.com/OpenGeoscience/geojs/pull/395#discussion_r31553930>:
>
> > +      .style({
> > +        value: function (d) { return d.z > -9999 ? d.z : null; }
> > +        /* You can get better contours if you set a minimum value and set
> > +         * sea locations to a small negative number:
> > +        value: function (d) { return d.z > -9999 ? d.z : -10; }
> > +         */
> > +      });
> > +    } else {
> > +      contour
> > +      .style({
> > +        value: function (d) { return d > -9999 ? d : null; }
> > +      })
> > +      .contour({
> > +        /* The geometry can be specified using 0-point coordinates and deltas
> > +         * since it is a regular grid. */
> > +        x0: data.x0, y0: data.y0, dx: data.dx, dy: data.dy
>
> so x0 and y0 are new keywords that contour accepts?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/OpenGeoscience/geojs/pull/395/files#r31553930>.
>



-- 
David Manthey
R&D Engineer
Kitware Inc.
(518) 881-4439